### DISCUSSION TITLE
How to retrieve a plugin object and its attributes from Jinja templates?

### DISCUSSION THREAD

**Discussion by Guts** [mkdocs/mkdocs#3776] [Q&A]
Hello,

I'm trying to adapt a template on my website to the configuration of a plugin, but I can't get the object's attributes from jinja files as I would like to do:

```jinja2
{% for plugin in config.plugins %}
  <plugin name="{{ plugin }}" />

  {% if plugin[:3] == "rss" and plugin.enabled == true %}
    <link rel="alternate" type="application/feed+json" title="{{ plugin.feed_title }} - Created"
    href="{{ 'plugin.feeds_filenames.json_created' | url }}" />
    <link rel="alternate" type="application/feed+json" title="{{ plugin.feed_title }} - Updated"
    href="{{ 'plugin.feeds_filenames.json_updated' | url }}" />
  {% endif %}

{% endfor %}
```

I had a look to <https://www.mkdocs.org/dev-guide/themes/#global-context> and I see that PluginsCollection is not exposed. Am I missing something?

**Comment by kamilkrzyskow** [comment]
Hi @Guts,
`plugin` in your context is a `key` string as seen with `plugin[:3] == "rss"`, but then you check `plugin.enabled`, and `str` doesn't have `enabled`. Funny that it doesn't crash (I haven't checked)

I recommend using `for name, instance in config.plugins.items()` to clearly separate strings from objects.
Also I typically read `enabled` from the config `instance.config.enabled`, not directly from the instance.

> I see that PluginsCollection is not exposed

It says:
> While you can use **any** config option, some commonly used options include:

and there is an example below:
> `{% if 'search' in config.plugins %}`
